app/src/androidTest/java/sdp/moneyrun/database/DatabaseProxyTest.java

```diff
@@ -39,38 +38,34 @@ public static void setPersistence(){
     }
     @Test
     public void getPlayerFromDatabase() {
-     //   FirebaseDatabase.getInstance().goOffline();
+       FirebaseDatabase.getInstance().goOffline();
         final Player player = new Player(1236, "Johann", 0);
         final PlayerDatabaseProxy db = new PlayerDatabaseProxy();
         CountDownLatch updated = new CountDownLatch(1);
         //adding to db
         CountDownLatch added = new CountDownLatch(1);
         OnCompleteListener listener = task -> added.countDown();
         db.putPlayer(player, listener);
-        try {
-            added.await(ASYNC_CALL_TIMEOUT, TimeUnit.SECONDS);
-            assertThat(added.getCount(), is(0L));
-        }catch (InterruptedException e){
-            fail();
-        }
-
         Task<DataSnapshot> testTask = db.getPlayerTask(player.getPlayerId());
         testTask.addOnCompleteListener(task -> {
             if(task.isSuccessful()) {
                assert( player.equals(db.getPlayerFromTask(testTask)));
                updated.countDown();
             }else{
+                System.out.println(task.getException());
                 assert (false);
             }
         });
-
+        while(!testTask.isComplete()){
+            System.out.println("false");
+        }
         try {
             updated.await(ASYNC_CALL_TIMEOUT, TimeUnit.SECONDS);
             assertEquals(0l, updated.getCount());
         } catch (InterruptedException e) {
             fail();
         }
-        //   FirebaseDatabase.getInstance().goOnline();
+          FirebaseDatabase.getInstance().goOnline();
```

Comment:
Same here, I get there is a reason why we go offline and then go online again, but don't really see it.

Reply:
At first it was to test if it was possible, and then with db rules having changed, it was so that permission wasn't denied

Reply:
Alright Alright!